vhost-user-backend/src/backend.rs

```diff
@@ -667,6 +691,11 @@ pub mod tests {
             GuestMemoryMmap::<()>::from_ranges(&[(GuestAddress(0x100000), 0x10000)]).unwrap(),
         );
         backend.update_memory(mem).unwrap();
+
+        backend.reset_device();
+        assert!(backend.lock().unwrap().events == 0);
+        assert!(!backend.lock().unwrap().event_idx);
+        assert!(backend.lock().unwrap().acked_features == 0);
```

Comment:
True. After some consideration, I think I’d rather leave it as-is, though: Given that this is a test, not a hot path, execution time is not a problem, but more the fact that `.lock().unwrap()` clutters the code visually (I think).

We are at the end of the scope here, so in theory we could just lock `backend`, but I wouldn’t find that good style, because if at some point code is added afterwards, it will need to take care to limit the lock’s scope again. So we should rather limit the scope here, too.

But if I add a new scope, e.g. like
```rust
{
    let backend = backend.lock().unwap();
    assert!(backend.events == 0);
    ...
}
```
or
```rust
assert!({
    let backend = backend.lock().unwrap();
    backend.events == 0 && !backend.event_idx && backend.acked_features == 0
})
```
I don’t really find that more readable than just taking the lock every time anymore.

(Of course, I also wouldn’t mind, though, if you find any of those more pleasant to read. :slightly_smiling_face:)

Reply:
> We are at the end of the scope here, so in theory we could just lock backend, but I wouldn’t find that good style, because if at some point code is added afterwards, it will need to take care to limit the lock’s scope again.

I was thinking about doing it that way. But I do not mind too much (I approved after all :P). Happy to leave it is as it is.

Reply:
Got it!